### PR TITLE
DWR-398 Modify stubbed data ingest to be safe for parallel processing

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: reporting/sql/V201706011496359299__item_claim_traget_not_null_ddl.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/**
+* change migrate update column to be refreshed on update
+**/
+
+USE ${schemaName};
+
+ALTER TABLE item MODIFY COLUMN claim_id smallint NOT NULL;
+ALTER TABLE item MODIFY COLUMN target_id smallint NOT NULL;
```
Comment (agorina) [inline reporting/sql/V201706011496359299__item_claim_traget_not_null_ddl.sql]: unrelated change.

File: warehouse/sql/V201702061486427077__initial_ddl.sql [hunk 2]
```diff
@@ -122,8 +122,7 @@ CREATE TABLE IF NOT EXISTS asmt (
   CONSTRAINT fk__asmt__import FOREIGN KEY (import_id) REFERENCES import(id),
   CONSTRAINT fk__asmt__update_import FOREIGN KEY (update_import_id) REFERENCES import(id)
 );
--- TODO: revisit this index
-ALTER TABLE asmt ADD INDEX idx__asmt_imports_deleted (import_id, update_import_id, deleted);
```
Comment (agorina) [inline warehouse/sql/V201702061486427077__initial_ddl.sql]: unrelated again. I removed all these indexes with TODOs and plan on revisiting them later.

File: warehouse/sql/V201702061486427077__initial_ddl.sql [hunk 3]
```diff
@@ -187,8 +186,8 @@ CREATE TABLE IF NOT EXISTS math_practice (
 
 CREATE TABLE IF NOT EXISTS item (
   id int NOT NULL AUTO_INCREMENT PRIMARY KEY,
-  claim_id smallint, -- TODO: change this to NOT NULL when we do not need a hack code that creates assmt
-  target_id smallint, -- TODO: change this to NOT NULL  when we do not need a hack code that creates assmt
+  claim_id smallint NOT NULL,
+  target_id smallint NOT NULL,
   natural_id varchar(40) NOT NULL,
```
Comment (agorina) [inline warehouse/sql/V201702061486427077__initial_ddl.sql]: Since the hack code does not seem to go away in the immediate future I am faking the claim_id and target_id and removed this TODO.

File: warehouse/sql/V201702061486427077__initial_ddl.sql [hunk 4]
```diff
@@ -567,3 +556,97 @@ CREATE PROCEDURE school_upsert(IN  p_district_name       VARCHAR(100),
...
+                                      IN  p_import_id   BIGINT,
+                                      OUT p_id          INT)
+  BEGIN
+
+    DECLARE isUpdate TINYINT;
+
+    --  handle duplicate entry: if there are two competing inserts, one will end up here
+    DECLARE CONTINUE HANDLER FOR 1062
+    BEGIN
+      SELECT id INTO p_id FROM student_group WHERE name = p_name AND school_id = p_school_id AND school_year = p_school_year;
```
Comment (agorina) [inline warehouse/sql/V201702061486427077__initial_ddl.sql]: This was the whole point of the change.

File: warehouse/sql/V201702061486427077__initial_ddl.sql [hunk 5]
```diff
@@ -567,3 +556,97 @@ CREATE PROCEDURE school_upsert(IN  p_district_name       VARCHAR(100),
...
+                                  IN  p_import_id     BIGINT,
+                                  OUT p_id            INT)
+  BEGIN
+
+    DECLARE isUpdate TINYINT;
+
+    --  handle duplicate entry: if there are two competing inserts, one will end up here
+    DECLARE CONTINUE HANDLER FOR 1062
+    BEGIN
+      SELECT id INTO p_id FROM asmt WHERE natural_id = p_natural_id;
```
Comment (agorina) [inline warehouse/sql/V201702061486427077__initial_ddl.sql]: And this is the part of the JIRA.

File: reporting/sql/V201706011496359299__item_claim_traget_not_null_ddl.sql (hunk 1 above)
Comment (malaffoon) [inline reporting/sql/V201706011496359299__item_claim_traget_not_null_ddl.sql]: typo in name of file, s/traget/target

Don't you have to put data in the columns first, or are they all populated?

Comment (agorina) [inline reporting/sql/V201706011496359299__item_claim_traget_not_null_ddl.sql]: You are right. Will add the script. Thank you!

File: reporting/sql/V201706011496359299__item_claim_traget_not_null_ddl.sql [hunk 6]
```diff
@@ -4,7 +4,8 @@
 
 USE ${schemaName};
 
--- please note that this is not instantenious, expect some delay..in seconds
+-- please note that this is not instantaneous, expect some delay..in seconds.
+-- This is temporary so I did not try to optimize it
 UPDATE item i
   JOIN asmt a
 SET
```
Comment (agorina) [inline reporting/sql/V201706011496359299__item_claim_traget_not_null_ddl.sql]: @esotrope and @bgsavage, just FYI: see note above.
